playwright/e2e/crypto/crypto.spec.ts

```diff
@@ -84,123 +77,73 @@ test.describe("Cryptography", function () {
-
-                // Verify that the SSSS keys are in the account data stored in the server
-                await verifyKey(app, "master");
-                await verifyKey(app, "self_signing");
-                await verifyKey(app, "user_signing");
-            });
-        });
+    /**
+     * Verify that the `m.cross_signing.${keyType}` key is available on the account data on the server
+     * @param keyType
+     */
+    async function verifyKey(app: ElementAppPage, keyType: "master" | "self_signing" | "user_signing") {
+        const accountData: { encrypted: Record<string, Record<string, string>> } = await app.client.evaluate(
+            (cli, keyType) => cli.getAccountDataFromServer(`m.cross_signing.${keyType}`),
+            keyType,
+        );
+
+        expect(accountData.encrypted).toBeDefined();
+        const keys = Object.keys(accountData.encrypted);
+        const key = accountData.encrypted[keys[0]];
+        expect(key.ciphertext).toBeDefined();
+        expect(key.iv).toBeDefined();
+        expect(key.mac).toBeDefined();
     }
 
+    test("Setting up key backup by recovery key", async ({ page, app, user: aliceCredentials }) => {
+        // Verified the device
+        await app.client.bootstrapCrossSigning(aliceCredentials);
+
+        await enableKeyBackup(app);
+
+        // Wait for the cross signing keys to be uploaded
+        // Waiting for "Change the recovery key" button ensure that all the secrets are uploaded and cached locally
+        const encryptionTab = await app.settings.openUserSettings("Encryption");
+        await expect(encryptionTab.getByRole("button", { name: "Change recovery key" })).toBeVisible();
+
+        // Verify that the SSSS keys are in the account data stored in the server
+        await verifyKey(app, "master");
+        await verifyKey(app, "self_signing");
+        await verifyKey(app, "user_signing");
+    });
+
     test("Can reset cross-signing keys", async ({ page, app, user: aliceCredentials }) => {
         await app.client.bootstrapCrossSigning(aliceCredentials);
         const secretStorageKey = await enableKeyBackup(app);
 
         // Fetch the current cross-signing keys
         async function fetchMasterKey() {
             return await test.step("Fetch master key from server", async () => {
-                const k = await app.client.evaluate(async (cli) => {
-                    const userId = cli.getUserId();
-                    const keys = await cli.downloadKeysForUsers([userId]);
-                    return Object.values(keys.master_keys[userId].keys)[0];
-                });
-                console.log(`fetchMasterKey: ${k}`);
+                const k = await app.client.evaluate(
+                    // @ts-ignore we can't use the CrossSigning enum here
+                    async (cli) => await cli.getCrypto().getCrossSigningKeyId("master"),
+                );
                 return k;
             });
```

Comment:
This no longer fetches the master key from the server, which (a) means the description on the test step is wrong, but more to the point, (b) rather defeats the point of the test.

Reply:
Hmm, I have the issue that `cli.downloadKeysForUsers` still returns the old master key. Cf [crypto lobby](https://matrix.to/#/!tahqRKMuWfcFOyQiPw:matrix.org/$yG1OXSuM9hO435bnf9UscMjrfWUoekytNwDddGCzJjk?via=matrix.org&via=element.io&via=one.ems.host) discussion, I wonder what we should do here.

Reply:
Due to https://github.com/matrix-org/matrix-js-sdk/pull/4772